packages/swap/src/routes/quote.ts

```diff
@@ -84,167 +101,147 @@ const quote = (io: Server) => {
-          asset: srcChainAsset.asset,
-          amount: boostFee.toString(),
-        });
+        includedFees.push(buildFee(srcAsset, 'BOOST', boostFee));
         swapInputAmount -= boostFee;
       }
 
-      let ingressFee = await getIngressFee(srcChainAsset);
+      let ingressFee = await getIngressFee(srcAsset);
       if (ingressFee == null) {
-        throw ServiceError.internalError(
-          `could not determine ingress fee for ${getInternalAsset(srcChainAsset)}`,
-        );
+        throw ServiceError.internalError(`could not determine ingress fee for ${srcAsset}`);
       }
       if (ingressEgressFeeIsGasAssetAmount) {
-        ingressFee = await estimateIngressEgressFeeAssetAmount(
-          ingressFee,
-          getInternalAsset(srcChainAsset),
-        );
+        ingressFee = await estimateIngressEgressFeeAssetAmount(ingressFee, srcAsset);
       }
-      includedFees.push({
-        type: 'INGRESS',
-        chain: srcChainAsset.chain,
-        asset: srcChainAsset.asset,
-        amount: ingressFee.toString(),
-      });
+      includedFees.push(buildFee(srcAsset, 'INGRESS', ingressFee));
       swapInputAmount -= ingressFee;
       if (swapInputAmount <= 0n) {
         throw ServiceError.badRequest(`amount is lower than estimated ingress fee (${ingressFee})`);
       }
 
       if (query.brokerCommissionBps) {
         const brokerFee = getPipAmountFromAmount(swapInputAmount, query.brokerCommissionBps);
-        includedFees.push({
-          type: 'BROKER',
-          chain: srcChainAsset.chain,
-          asset: srcChainAsset.asset,
-          amount: brokerFee.toString(),
-        });
+        includedFees.push(buildFee(srcAsset, 'BROKER', brokerFee));
         swapInputAmount -= brokerFee;
       }
 
-      const quotePools = await getPools(
-        getInternalAsset(srcChainAsset),
-        getInternalAsset(destChainAsset),
+      const poolQuote = getPoolQuoteResult(
+        srcAsset,
+        destAsset,
+        swapInputAmount,
+        ingressEgressFeeIsGasAssetAmount,
+        includedFees,
+      );
+
+      const pools = await getPools(srcAsset, destAsset);
+
+      const firstLegPoolFee = getHundredthPipAmountFromAmount(
+        swapInputAmount,
+        pools[0].liquidityFeeHundredthPips,
       );
 
-      const quoteRequest = buildQuoteRequest({
-        ...query,
-        amount: String(swapInputAmount),
-      });
-      io.emit('quote_request', quoteRequest);
+      includedFees.push(buildFee(srcAsset, 'LIQUIDITY', firstLegPoolFee));
+
+      if (!quoter.canQuote()) {
+        const result = await poolQuote;
+
+        logger.info('sending pool quote', {
+          USE_JIT_QUOTING: env.USE_JIT_QUOTING,
+          connectedSockets: io.sockets.sockets.size,
+          success: result.success,
+          response: result.success ? result.data : result.reason,
+        });
+
+        if (result.success) {
+          res.json({ ...result.data, quoteType: undefined });
```

Comment:
is `quoteType: undefined` intentional?

Reply:
Yeah it overwrites it so when it gets stringified, the key is deleted